pkg/manifests/external_dns.go

```diff
@@ -272,6 +272,17 @@ func newExternalDNSDeployment(conf *config.Config, externalDnsConfig *ExternalDn
 								corev1.ResourceMemory: resource.MustParse("250Mi"),
 							},
 						},
+						SecurityContext: &corev1.SecurityContext{
+							Privileged:               util.ToPtr(false),
+							AllowPrivilegeEscalation: util.ToPtr(false),
+							ReadOnlyRootFilesystem:   util.ToPtr(true),
+							RunAsNonRoot:             util.ToPtr(true),
+							RunAsUser:                util.Int64Ptr(65532),
```

Comment:
is this specific uid documented by externalDNS anywhere or is it arbitrary? wonder if there's a specific one we should use for the same reasons that we discussed during the operator securitycontext PR

Reply:
this is what's recommended by the external dns team. it's a generic non root